src/server/graphics/multiplexing_hw_cursor.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "multiplexing_hw_cursor.h"
+
+#include "mir/graphics/display.h"
+#include <boost/throw_exception.hpp>
+
+namespace mg = mir::graphics;
+
+namespace
+{
+auto construct_platform_cursors(std::span<mg::Display*> platform_displays) -> std::vector<std::shared_ptr<mg::Cursor>>
+{
+    std::vector<std::shared_ptr<mg::Cursor>> cursors;
+    for (auto display : platform_displays)
+    {
+        cursors.push_back(display->create_hardware_cursor());
+        if (cursors.back() == nullptr)
+        {
+            BOOST_THROW_EXCEPTION((std::runtime_error{"Platform failed to create hardware cursor"}));
+        }
```

Comment:
Probably not a problem here, but thinking forward: this means that when adding a virtual display we offer different behaviour. (Obviously, fixing that would need something more complex than "hardware cursor" xor "software cursor")